examples/basics/demo_phase_retrieval.py

```diff
@@ -0,0 +1,276 @@
+plt.show()
+
+# %%
+# Phase correction and signal reconstruction
+# -----------------------------------------------------------
+
+# correct possible global phase shifts
+x_gd_spec = correct_global_phase(x_phase_gd_spec, x_phase)
+# extract phase information and normalize to the range [0, 1]
+x_gd_spec = torch.angle(x_gd_spec) / torch.pi + 0.5
+plot([x, x_gd_spec], titles=["Signal", "Reconstruction"], rescale_mode="clip")
+
+# %%
+# Reconstruction with gradient descent and PnP denoisers
+# ---------------------------------------------------------------
+# We can also use the Plug-and-Play (PnP) framework to incorporate denoisers as regularizers in the optimization algorithm. We use a deep denoiser as the prior, which is trained on a large dataset of natural images.
+
+# Load the pre-trained denoiser
+denoiser = DRUNet(
+    in_channels=n_channels,
+    out_channels=n_channels,
+    pretrained="download",  # automatically downloads the pretrained weights, set to a path to use custom weights.
+    train=False,
+    device=device,
+)
+# The original denoiser is designed for real-valued images, so we need to convert it to a complex-valued denoiser for phase retrieval problems.
+denoiser_complex = to_complex_denoiser(denoiser)
+
+# Algorithm parameters
+data_fidelity = L2()
+prior = PnP(denoiser=denoiser_complex)
+params_algo = {"stepsize": 0.10, "g_param": 0.05}
+max_iter = 100
+early_stop = True
+verbose = True
+
+# Instantiate the algorithm class to solve the IP problem.
+model = optim_builder(
+    iteration="PGD",
+    prior=prior,
+    data_fidelity=data_fidelity,
+    early_stop=early_stop,
+    max_iter=max_iter,
+    verbose=verbose,
+    params_algo=params_algo,
+)
+
+# Run the algorithm
+x_phase_pnp, metrics = model(y, physics, x_gt=x_phase, compute_metrics=True)
+
+# %%
+# Phase correction and signal reconstruction
+# -----------------------------------------------------------
+
+# correct possible global phase shifts
+x_pnp = correct_global_phase(x_phase_pnp, x_phase)
+# now no global phase shift should exist
+# assert torch.allclose(x_est, x_phase)
+# extract phase information and normalize to the range [0, 1]
+x_pnp = torch.angle(x_pnp) / (2 * torch.pi) + 0.5
+plot([x, x_pnp], titles=["Signal", "Reconstruction"], rescale_mode="clip")
+
+# %%
+# Overall comparison
+# -----------------------------------------------------------
+# We visualize the original image and the reconstructed images from the four methods.
+# We further computed the PSNR (Peak Signal-to-Noise Ratio) scores (higher better) for every reconstruction and their cosine similarities with the original image (range in [0,1], higher better).
+# In conclusion, gradient descent with ranodom intialization barely provides a reconstruction, while spectral methods provide a good initial estimate which can later be improved by gradient descent. Besides, the PnP framework with a deep denoiser as the prior provides the best denoising effect and overall reconstruction results.
```

Comment:
"ranodom intialization" -> "random initialization"
"barely provides a reconstruction" -> "provides a poor reconstruction"

Reply:
"the best denoising effect and overall reconstruction results." -> "provides the best denoising results as it exploits prior information about the set of natural images"